File: day2/function_prototype.rb
class Person
  def initialize(name)
    @name = name
  end
  def show_name
    puts @name
  end

  def say_hi
    puts hi
  end
end

person1 = Person.new("andrew")
person2 = Person.new("Long")
person1.show_name
person2.show_name

File: day2/function_prototype.js
function Human(firstName, lastName) {
  this.firstName = firstName;
  this.lastName = lastName;
  this.fullname = function() {
    return `${this.firstName} ${this.lastName}`;
  }
}

// function instance
person1 = new Human("Alex", "Holder");
person2 =new Human("Oliver", "Hardman");
Human.prototype.name = "Arthur";
console.log(person1);
console.log(Human.prototype);
console.log(person1.name);
console.log(person2.name);
